### PR TITLE
Worldengine

### PR DESCRIPTION
**Pull request by rardiol**
###### Motivation for this change

google_apputils seems to work on py3k as is? Why is it disabled? I also am not sure yet if ignoring the strict version requirements is advisable.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -12171,7 +12171,6 @@ in {
 
   google_apputils = buildPythonPackage rec {
     name = "google-apputils-0.4.1";
-    disabled = isPy3k;
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: probably it was incompatible in older versions

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: d568cb971be3c1a9f732e2d254baf74dff84fa2a

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -27805,6 +27805,36 @@ in {
...
+    name = "worldengine-${version}";
+    version = "0.19.0";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/w/worldengine/${name}.tar.gz";
+      sha256 = "0wnpp0zm4idc4242a970yf1blf8y2f1ikwha0q781wfz1dg4z0i6";
+    };
+
+    propagatedBuildInputs = with self; [ noise numpy pyplatec protobuf3_1 purepng argparse h5py gdal ];
+    doCheck = true;
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: this flag is implicit for `doCheck`

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -27805,6 +27805,36 @@ in {
     };
   };
 
+  worldengine = buildPythonPackage rec {
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: Is this a library or just a command line application?

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -27805,6 +27805,36 @@ in {
...
+
+    prePatch = ''
+      substituteInPlace setup.py \
+        --replace pypng>=0.0.18 purepng \
+        --replace 'numpy>=1.9.2, <= 1.10.0.post2' 'numpy' \
+        --replace 'argparse==1.2.1' 'argparse' \
+        --replace 'protobuf==3.0.0a3' 'protobuf' \
+    '';
+      # --replace 'noise==1.2.2' 'noise' \
+      # --replace 'PyPlatec==1.4.0' 'PyPlatec' \
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: what is with the two commented lines above?

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: oh, I removed it actually :)

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (rardiol) [inline pkgs/top-level/python-packages.nix]: command line application

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (rardiol) [inline pkgs/top-level/python-packages.nix]: I wrote the code to drop all specific version, but those 2 are actually the versions on nixpkgs, so I commented them out. I guess it's pointless now, will remove.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (rardiol) [inline pkgs/top-level/python-packages.nix]: ok, removed

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: No replace them as well. It's bad practice to pin exact versions, and its quite likely in the future these specific versions are not available.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (rardiol) [inline pkgs/top-level/python-packages.nix]: Does the duplicated removal generate a conflict and I need to remove it or git handles it?

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (rardiol) [inline pkgs/top-level/python-packages.nix]: Ok, done.

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: In this case we usually put them outside of `python-packages`, because we only need one version of it.

**Comment by rardiol** [comment]
Why does it build gdal instead of fetching from the store?

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: no in case its ok, to have in pythonPackages.

**Comment by jpierre03** [comment]
(triage) @rardiol Is there any update ?

**Comment by rardiol** [comment]
Rebased it on origin/master, still works. The requirements.txt seems to be still open. I just wanted to know if I need to declare a worldengine-cli on all-packages.nix so hydra caches it. gdal particularly  takes a while to compile.

**Comment by FRidh** [comment]
Are the tests running?

**Comment by rardiol** [comment]
I just tried setting doCheck = true; and it shows 0 tests as being run. There's a tests directory and a tox.ini file, so I assume that means it uses tox?

**Comment by rardiol** [comment]
Enabled testing.

**Pull request merged**